File: 프로그래머스/1/388351. 유연근무제/유연근무제.js
function solution(schedules, timelogs, startday) {
  let events = 0;
  const MAX_MINUTES = 11 * 60 + 59; // 11:59을 초과하는 로그는 무시

  for (let i = 0; i < timelogs.length; i++) {
    const schedule = schedules[i];
    // schedule: HHMM 형태라 가정
    let scheduleMinutes = Math.floor(schedule / 100) * 60 + (schedule % 100);
    let cutoffMinutes = scheduleMinutes + 10;
    if (cutoffMinutes > MAX_MINUTES) cutoffMinutes = MAX_MINUTES;

    let ontimeCount = 0;
    const logs = timelogs[i] || [];

    for (let j = 0; j < logs.length; j++) {
      // startday: 1=월요일 ... 7=일요일 이므로, j(0..n) 기준으로 weekday(1..7) 계산
      const weekday = ((startday - 1 + j) % 7) + 1;
      if (weekday === 6 || weekday === 7) continue; // 토(6), 일(7) 제외

      const log = logs[j];
      if (log == null) continue; // 결석/빈값 처리

      const logHour = Math.floor(log / 100);
      const logMinute = log % 100;
      const logTotal = logHour * 60 + logMinute;

      if (logTotal > MAX_MINUTES) continue; // 11:59 이후 무시

      if (logTotal <= cutoffMinutes) ontimeCount++;
    }

    if (ontimeCount >= 5) events++;
  }

  return events;
}

File: 프로그래머스/1/388351. 유연근무제/유연근무제.py
def solution(schedules, timelogs, startday):
    count = 0
    # 희망 출근 시간과 실제 출근 시간 배열과 매칭 
    for s, t in zip(schedules, timelogs):
        # 희망 출근 시간 이전에 출근 여부 확인 
        logs = [False, False, False, False, False]
        for i, d in enumerate(t):
            # 토요일 일요일은 확인 범위에서 제외 
            if (i + startday) % 7 == 6 or (i + startday) % 7 == 0:
                continue
                
            # 출근 예상 시간 구하기 
            expect = s + 10
            hour = expect // 100
            minute = expect % 100
            if minute >= 60:
                minute = minute % 60
                hour += 1
                expect = hour * 100 + minute
            
            # 희망 출근 시간 이전에 출근한 경우 확인 
            if d <= expect:
                logs[((i + startday) % 7) - 1] = True
        
        if all(logs):
            count += 1
    return count